### ISSUE TITLE
[Feature] Process complex parameters based on the value of @Step annotation

### ISSUE THREAD

**Issue by a-en** [serenity-bdd/serenity-core#1922]
Here is the use case - a bunch of steps methods have objects as parameters (e.g. `doAction(User user)`). By default, those objects are represented in the report as a result of the invocation of `toString()` method. Taking into account those objects are pretty complex (they have a bunch of properties) output is pretty huge. Unfortunately, in this case, it is not possible to override `toString()` method. 

One of the possible solutions is to implement additional logic, that would allow providing additional instructions on how those parameters should be reflected in the report.

For example, in addition to the currently supported reference to the particular parameter by its index, we could refer to some properties or methods of that parameter.
So in addition to 

```
@Step("Open as user {0}")
public void open(User user)
```

we could have something like

```
@Step("Open as user {0.email}")
public void open(User user)
```
or
```
@Step("Open as user {0.getEmail()}")
public void open(User user)
```

It would require changes in `ExecutedStepDescription` to store raw arguments and then in `AnnotatedStepDescription` via reflection we could resolve the desired representation.

@wakaleo, I'm ready to provide PR if you think it might be added in that way.

**Comment by a-en** [comment]
@wakaleo any thoughts on that?

**Comment by wakaleo** [comment]
Excellent idea I had considered something similar at some point but never implemented it.

**Comment by wakaleo** [comment]
Implemented in 2.1.5
